### ISSUE TITLE
NPE from shadow models plugin

### ISSUE THREAD

**Issue by ftomassetti** [JetBrains/MPS-extensions#249]
Hi, I am experiencing issues building languages since updating the mbeddr platform. In particular I get this exception:

```
Failed to dispose part class de.q60.mps.shadowmodels.runtime.plugin.ProjectPlugin_ProjectPluginPart of project plugin class de.q60.mps.shadowmodels.runtime.plugin.Runtime_ProjectPlugin

java.lang.NullPointerException
	at de.q60.mps.shadowmodels.runtime.plugin.ProjectViewExtension.dispose(ProjectViewExtension.java:199)
	at de.q60.mps.shadowmodels.runtime.plugin.ProjectPlugin_ProjectPluginPart.dispose(ProjectPlugin_ProjectPluginPart.java:17)
	at jetbrains.mps.plugins.projectplugins.BaseProjectPlugin.disposeCustomPluginParts(BaseProjectPlugin.java:163)
	at jetbrains.mps.plugins.projectplugins.BaseProjectPlugin.disposePrefsToolsAndCustomParts(BaseProjectPlugin.java:155)
	at jetbrains.mps.plugins.projectplugins.BaseProjectPlugin.dispose(BaseProjectPlugin.java:144)
	at jetbrains.mps.plugins.projectplugins.ProjectPluginManager.disposePlugin(ProjectPluginManager.java:189)
	at jetbrains.mps.plugins.projectplugins.ProjectPluginManager.disposePlugin(ProjectPluginManager.java:57)
	at jetbrains.mps.plugins.BasePluginManager.disposePlugins(BasePluginManager.java:165)
	at jetbrains.mps.plugins.BasePluginManager.unloadPlugins(BasePluginManager.java:113)
	at jetbrains.mps.plugins.PluginLoaderRegistry.unloadContributors(PluginLoaderRegistry.java:223)
	at jetbrains.mps.plugins.PluginLoaderRegistry$UpdatingTask.removeContributors(PluginLoaderRegistry.java:453)
	at jetbrains.mps.plugins.PluginLoaderRegistry$UpdatingTask.doUpdate(PluginLoaderRegistry.java:415)
	at jetbrains.mps.plugins.PluginLoaderRegistry$UpdatingTask.lambda$run$0(PluginLoaderRegistry.java:402)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:312)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:152)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:433)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:416)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:399)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:824)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:412)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:693)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:411)
	at com.intellij.ide.IdeEventQueue.pumpEventsForHierarchy(IdeEventQueue.java:881)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:205)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runProcessWithProgressSynchronously(ApplicationImpl.java:566)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressSynchronously(CoreProgressManager.java:459)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.runProcessWithProgressSynchronously(ProgressManagerImpl.java:82)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runSynchronously(CoreProgressManager.java:324)
	at com.intellij.openapi.progress.impl.CoreProgressManager.run(CoreProgressManager.java:309)
	at jetbrains.mps.plugins.PluginLoaderRegistry.runTask(PluginLoaderRegistry.java:237)
	at jetbrains.mps.plugins.PluginLoaderRegistry.update(PluginLoaderRegistry.java:358)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:312)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:433)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:416)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:399)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:824)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:412)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:704)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:411)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by slisson** [comment]
This is already fixed for MPS 2019.2 (https://github.com/JetBrains/MPS-extensions/commit/84751b1a7b95a6e2ea795c8239655a3ab4d449b0)

**Comment by slisson** [comment]
Yes, your version is older than the fix.
